apis/deployer/healthchecks/validation/validation.go

```diff
@@ -0,0 +1,121 @@
+func ValidateCustomHealthCheckConfiguration(fldPath *field.Path, config *healthchecks.CustomHealthCheckConfiguration) field.ErrorList {
+	var allErrs field.ErrorList
+	if len(config.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must not be empty"))
+	}
+
+	if config.Resource == nil && config.LabelSelector == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("resourceSelector/labelSelector"), "must have either a resource or a set of labels"))
+	}
+
+	if config.LabelSelector != nil {
+		allErrs = append(allErrs, ValidateLabelSelectorSpec(fldPath.Child("labelSelector"), config.LabelSelector)...)
+	}
+
+	if len(config.Requirements) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("requirements"), "must have at least one resource requirement"))
+
+	}
+
+	for _, r := range config.Requirements {
+		allErrs = append(allErrs, ValidateRequirementSpec(fldPath.Child("requirements"), &r)...)
+	}
+
+	return allErrs
+}
+
+// ValidateRequirementSpec validates a requirement specification for a custom healthcheck configuration
+func ValidateRequirementSpec(fldPath *field.Path, spec *healthchecks.RequirementSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(spec.JsonPath) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("jsonPath"), "JSON path must be defined"))
+	}
+
+	p := jsonpath.New("test")
+	err := p.Parse(spec.JsonPath)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("jsonPath"), spec.JsonPath, "is not a valid JSON path template"))
+	}
+
+	op := spec.Operator
+	err = validateOperator(op)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("operator"), op, "is not a valid operator"))
+	}
+
+	if op != selection.DoesNotExist && op != selection.Exists && len(spec.Value) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("value"), "must have at least one value to compare to"))
+	}
+
+	if op != selection.In && op != selection.NotIn && len(spec.Value) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("value"), op, "must have exactly one value to compare to"))
+	}
+
+	return allErrs
+}
+
+// ValidateLabelSelectorSpec validates a LabelSelector specification for a custom healthcheck configuration
+func ValidateLabelSelectorSpec(fldPath *field.Path, spec *healthchecks.LabelSelectorSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(spec.Labels) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("matchLabels"), "must have at least one label to select on"))
+	}
+
+	return allErrs
+}
+
+func validateOperator(operator selection.Operator) error {
+	var allowedOperators = []selection.Operator{
+		selection.DoesNotExist,
+		selection.Exists,
+		selection.Equals,
+		selection.DoubleEquals,
+		selection.NotEquals,
+		selection.In,
+		selection.NotIn,
+	}
+
+	for _, op := range allowedOperators {
+		if operator == op {
+			return nil
+		}
+	}
```

Comment:
nit maybe you can use a switch statement. then the allocation and loop is not required
```suggestion
	var allowedOperators = []selection.Operator{
		selection.DoesNotExist,
		selection.Exists,
		selection.Equals,
		selection.DoubleEquals,
		selection.NotEquals,
		selection.In,
		selection.NotIn,
	}

	switch op {
	case selection.DoesNotExist, ...:
	  return nil
	 default:
	   return err
	 }
```